taskfiles/utils/cpp-lint.yaml

```diff
@@ -4,36 +4,315 @@ set: ["u", "pipefail"]
-        \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
-        -print0 | \
-          xargs -0 clang-format {{.FLAGS}} -Werror
+        {{- if .INCLUDE_PATTERNS}}
+          \( \
+            {{- range $i, $pattern := .INCLUDE_PATTERNS}}
+              {{- if $i}}
+                -or \
+              {{- end}}
+              -path "{{$pattern}}" \
+            {{- end}}
+          \) \
+        {{- end}}
+        {{- if .INCLUDE_FILENAME_PATTERNS}}
+          \( \
+            {{- range $i, $pattern := .INCLUDE_FILENAME_PATTERNS}}
+              {{- if $i}}
+                -or \
+              {{- end}}
+              -name "{{$pattern}}" \
+            {{- end}}
+          \) \
+        {{- end}}
+        -print0
+      | xargs -0 clang-format \
+        {{- range .FLAGS}}
+          {{.}} \
+        {{- end}}
+        -Werror
 
-  # Runs clang-tidy on C++ files at the given paths.
+  # Runs `clang-tidy` on `SOURCE_FILE_PATH` and writes the output to `OUTPUT_FILE_PATH`. The strings
+  # "Passed" or "Failed" are appended to the output file, depending on the result.
   #
-  # @param {string} FLAGS Any flags to pass to clang-tidy.
-  # @param {string[]} SRC_PATHS The paths on which to run clang-tidy.
-  # @param {string} VENV_DIR Python virtual environment where clang-tidy is installed.
+  # @param {string} OUTPUT_FILE_PATH
+  # @param {string} SOURCE_FILE_PATH
+  # @param {string} VENV_DIR Python virtual environment where `clang-tidy` is installed.
+  # @param {string[]} [FLAGS] Any flags to pass to `clang-tidy`.
   clang-tidy:
     internal: true
+    label: "{{.TASK}}:{{.SOURCE_FILE_PATH}}"
+    sources: ["{{.SOURCE_FILE_PATH}}"]
+    vars:
+      FLAGS:
+        ref: "default (list) .FLAGS"
     requires:
-      vars: ["FLAGS", "SRC_PATHS", "VENV_DIR"]
-    cmd: |-
-      . "{{.VENV_DIR}}/bin/activate"
-      find {{- range .SRC_PATHS}} "{{.}}" {{- end}} \
-        -type f \
-        \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
-        -print0 | \
-          xargs -0 clang-tidy {{.FLAGS}}
+      vars: ["OUTPUT_FILE_PATH", "SOURCE_FILE_PATH", "VENV_DIR"]
+    generates: ["{{.OUTPUT_FILE_PATH}}"]
+    cmds:
+      - defer: |-
+          if [[ 0 -eq "{{.EXIT_CODE}}" ]]; then
+            echo "Passed" >> "{{.OUTPUT_FILE_PATH}}"
+          else
+            echo "Failed" >> "{{.OUTPUT_FILE_PATH}}"
+          fi
+      - |-
+        . "{{.VENV_DIR}}/bin/activate"
+        clang-tidy \
+          {{- range .FLAGS}}
+            {{.}} \
+          {{- end}}
+          "{{.SOURCE_FILE_PATH}}" &> "{{.OUTPUT_FILE_PATH}}"
+
+  # Runs `clang-tidy` in parallel on C++ source files that differ between HEAD and the target
+  # branch. On failure, `clang-tidy`'s output is printed. The task will fail on the first file which
+  # fails `clang-tidy`'s checks, which will result in some files having incomplete output. Users can
+  # either fix the known errors and run the task again or directly run `clang-tidy` on a file of
+  # interest.
```

Comment:
Since `clang-tidy-find` is similar except that it works with path filtering, should we include the same details in its docstring (e.g., that clang-tidy is run in parallel)?

Reply:
Ya you're right. I was thinking users could read the called tasks, but that seems like an unnecessary burden.